control-plane/subcommand/acl-init/command.go

```diff
@@ -143,9 +159,28 @@ func (c *Command) Run(args []string) int {
 		}
 	}
 
+	var secret string
 	if c.flagACLAuthMethod != "" {
 		cfg := api.DefaultConfig()
 		c.http.MergeOntoConfig(cfg)
+
+		if len(c.flagServerAddresses) > 0 {
+			serverAddresses, err := common.GetResolvedServerAddresses(c.flagServerAddresses, c.providers, c.logger)
+			if err != nil {
+				c.UI.Error(fmt.Sprintf("Unable to discover any Consul addresses from %q: %s", c.flagServerAddresses[0], err))
+				return 1
+			}
+
+			scheme := "http"
+			if c.flagUseHTTPS {
+				scheme = "https"
+			}
+			// For all of the next operations we'll need a Consul client.
+			serverAddr := fmt.Sprintf("%s:%d", serverAddresses[0], c.flagServerPort)
+			cfg.Address = serverAddr
+			cfg.Scheme = scheme
```

Comment:
This whole added block is taken from prior art in `partition-init` command. basically an external server(s) has been passed in and we want to talk to that and not the default endpoint.